### DISCUSSION TITLE
Probability score normalization

### DISCUSSION THREAD

**Discussion by johanenforcer** [ultralytics/yolov5#2861] [Q&A]
I have noticed that the aggregate of the probability scores retrieved from bounding boxes do not sum up to 1 spefically the x[:, 5:] found in general.py. Can I then use lets say a sigmoid activation function on all of the probability scores for a certain bbox to normalize them for that particular box?

**Comment by glenn-jocher** [comment]
@johanenforcer yes this is correct. Starting in YOLOv3 the class probabilities are no longer normalized using softmax. This allows for multi-label inference, i.e. one box could have multiple labels like 'person' and 'woman'.

**Reply by johanenforcer** [reply]
Thank you! Btw, sorry if this is off-topic but how many generations are "required" to yield decent hyperparameters? Is it ok to reduce the number of 300 generations substantially and if so, what is the limit generally speaking which of course is very hard to say. I'm asking this because I think its not feasible for me to use the default settings.

**Reply by glenn-jocher** [reply]
@johanenforcer well there aren't many specific rules here as you can imagine. < 1 generation will not yield results and >0 will start to yield results, with improved results correlating with increasing generations up to a point when things will begin to plateau. The plateau point would be determined by your specific hyp initial conditions, dataset, model, fitness function etc. For YOLOv5m on VOC this plateau seems to start around 200-300 generations.

**Reply by johanenforcer** [reply]
I forgot to ask but I assume it would be fine to also use a softmax activation function on each bounding box found in the x[:, 5:] tensor, directly in for example the NMS function in general.py when running inference?

**Reply by glenn-jocher** [reply]
@johanenforcer yeah that would work fine! If you implement that take care to specify the softmax dimension correctly.

**Reply by johanenforcer** [reply]
Thanks!
